templates/server/src/main/resources/archetype-resources/core/src/test/java/__packageInPathFormat__/general/common/base/test/DbTestHelper.java

```diff
@@ -1,44 +1,76 @@
 package ${package}.general.common.base.test;
 
 import javax.inject.Named;
-
+#if ($dbMigration == 'liquibase')  
+ import liquibase.Liquibase;
+#else 
 import org.flywaydb.core.Flyway;
-
+#end
 /**
  * This class provides methods for handling the database during testing where resets (and other operations) may be
  * necessary.
  */
 @Named
 public class DbTestHelper {
-
-  private Flyway flyway;
-
+ #if ($dbMigration == 'liquibase')  
+   private Liquibase liquibase; 
+ #else 
+   private Flyway flyway;
+ #end
+ 
+ #if ($dbMigration == 'liquibase')  
+   /**
+    * The constructor.
+    *
+    * @param liquibase an instance of type {@link Liquibase}.
+    */
+   public DbTestHelper( Liquibase liquibase) {
+     super();
+       this.liquibase = liquibase;
+ }
+ #else 
+   /**
+    * The constructor.
+    *
+    * @param flyway an instance of type {@link Flyway}.
+    */
+   public DbTestHelper( Flyway flyway) {
+     super();
+       this.flyway = flyway;
+ }
+ #end
+ 
+  #if ($dbMigration == 'flyway')
   /**
-   * The constructor.
-   *
-   * @param flyway an instance of type {@link Flyway}.
+   * Calls {@link #dropDatabase()} internally, and migrates to the highest available migration (default) or to the
+   * {@code migrationVersion} specified by {@link #setMigrationVersion(String)}.
    */
-  public DbTestHelper(Flyway flyway) {
-    super();
-    this.flyway = flyway;
+  public void resetDatabase() {
+    
+      this.flyway.migrate();   
   }
+  #end
 
+#if ($dbMigration == 'liquibase')  
   /**
    * Drops the whole database.
    */
-  public void dropDatabase() {
+  public void cleanup() {
 
-    this.flyway.clean();
-  }
-
-  /**
-   * Calls {@link #dropDatabase()} internally, and migrates to the highest available migration (default) or to the
-   * {@code migrationVersion} specified by {@link #setMigrationVersion(String)}.
-   */
-  public void resetDatabase() {
+    try {
+      this.liquibase.dropAll();
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
```

Comment:
same as above. We should take the opportunity to use `TestCleanerPlugin` and avoid any redundant switches between flyway or liquibase in code like this.

Reply:
Implemented testcleanerPlugin here also to get rid of redundant switches